### PR TITLE
testing/nginx-naxsi: update to 1.11.8 with dynamic modules

### PR DESCRIPTION
**Pull request by itoffshore**
dynamic module support added

`/usr/lib/nginx/modules`:

3RD PARTY
================================

```
ngx_http_naxsi_module.so
ngx_http_cache_purge_module.so
ngx_http_upstream_fair_module.so
ngx_http_sysguard_module.so
```

* http-sysguard backported from [tengine](http://tengine.taobao.org/document/http_sysguard.html).

CORE
================================
```
ngx_http_geoip_module.so
ngx_http_image_filter_module.so
ngx_http_perl_module.so
ngx_http_xslt_filter_module.so
ngx_mail_module.so
ngx_stream_module.so
```

### CODE REVIEW

File: testing/nginx-naxsi/APKBUILD [hunk 1]
```diff
@@ -5,25 +5,44 @@
...
+_ngx_upstream_fair_dir="$srcdir/$_ngx_upstream_fair_name-$_ngx_upstream_fair_ver"
+
+_ngx_http_sysguard_name=tengine-http-sysguard
+_ngx_http_sysguard_ver=2.2.0
+_ngx_http_sysguard_dir="$srcdir/$_ngx_http_sysguard_name-$_ngx_http_sysguard_ver"
+
 depends="!nginx"
-makedepends="pcre-dev libressl-dev zlib-dev paxmark linux-headers"
-subpackages="$pkgname-doc $pkgname-vim:vim"
+makedepends="linux-headers gd-dev geoip-dev libxml2-dev libxslt-dev luajit-dev
```
Comment (jirutka) [inline testing/nginx-naxsi/APKBUILD]: Why `luajit-dev` when there’s no Lua module…?

File: testing/nginx-naxsi/APKBUILD [hunk 2]
```diff
@@ -32,119 +51,154 @@ source="http://nginx.org/download/$_pkgname-$pkgver.tar.gz
...
+	cd "$builddir"
 	make DESTDIR="$pkgdir" install
 
-	local paxflags="-m"
-	[ "$CARCH" = "x86" ] && paxflags="-msp"
-	paxmark "$paxflags" "$pkgdir"/usr/sbin/nginx || return 1
+	# Disable some PaX protections; this is needed for Lua module.
+#	local paxflags="-m"
+#	[ "$CARCH" = "x86" ] && paxflags="-msp"
+#	paxmark "$paxflags" "$pkgdir"/usr/sbin/nginx || return 1
```
Comment (jirutka) [inline testing/nginx-naxsi/APKBUILD]: If the code is not needed anymore, delete it, do not comment it out…

File: testing/nginx-naxsi/APKBUILD [hunk 3]
```diff
@@ -32,119 +51,154 @@ source="http://nginx.org/download/$_pkgname-$pkgver.tar.gz
...
 
 	install -m755 -D "$srcdir"/$_pkgname.initd "$pkgdir"/etc/init.d/$_pkgname
 	install -m644 -D "$srcdir"/$_pkgname.logrotate "$pkgdir"/etc/logrotate.d/$_pkgname
 
 	install -m644 -D LICENSE "$pkgdir"/usr/share/licenses/$pkgname/LICENSE
 	install -m644 -D "$srcdir"/naxsi-$_ngx_naxsi_ver/naxsi_config/naxsi_core.rules "$pkgdir"/etc/nginx/naxsi_core.rules
+	mkdir -p "$pkgdir"/var/lib/nginx/tmp
```
Comment (jirutka) [inline testing/nginx-naxsi/APKBUILD]: Why? Temporary files should be stored in /tmp or /var/tmp, not /var/lib.

File: testing/nginx-naxsi/APKBUILD [hunk 4]
```diff
@@ -32,119 +51,154 @@ source="http://nginx.org/download/$_pkgname-$pkgver.tar.gz
...
 }
 
 vim() {
-        local t
+        pkgdesc="$pkgdesc (vim syntax)"
+        depends=
+
+        mkdir -p "$subpkgdir"/usr/share/vim
+        cp -r "$builddir"/contrib/vim "$subpkgdir"/usr/share/vim/vimfiles
+}
```
Comment (jirutka) [inline testing/nginx-naxsi/APKBUILD]: There’s already `nginx-vim` provided by the `main/nginx` package, so this subpackage is redundant.

File: testing/nginx-naxsi/nginx.initd [hunk 5]
```diff
@@ -1,42 +1,69 @@
...
+cfgfile=${cfgfile:-/etc/nginx/nginx.conf}
+pidfile=/run/nginx/nginx.pid
+tmpdir=/var/lib/nginx/tmp
+command=/usr/sbin/nginx
+command_args="-c $cfgfile"
+required_files="$cfgfile"
 
 depend() {
-	need net
+#	need net => breaks LXC containers ##
```
Comment (jirutka) [inline testing/nginx-naxsi/nginx.initd]: I don’t think so… I use LXC and don’t have such problem.

File: testing/nginx-naxsi/APKBUILD [hunk 6]
```diff
@@ -32,119 +51,154 @@ source="http://nginx.org/download/$_pkgname-$pkgver.tar.gz
...
+
+_module() {
+        local name="${subpkgname#$pkgname-mod-}"
+        name="${name//-/_}"
+        local soname="$(eval "echo \$_${name}_so")";
+        soname="${soname:-"ngx_${name}_module.so"}"
+
+        pkgdesc="$pkgdesc (module $name)"
+        depends="$pkgname $(eval "echo \$_${name}_depends")"
```
Comment (jirutka) [inline testing/nginx-naxsi/APKBUILD]: Some of the modules are already provided by the `main/nginx` package, so you must set `depends="!nginx-mod-$name"` for them.

Comment (jirutka) [inline testing/nginx-naxsi/APKBUILD]: You don’t declare any `_<mod>_depends` variable, so this is redundant.

File: testing/nginx-naxsi/APKBUILD [hunk 7]
```diff
@@ -32,119 +51,154 @@ source="http://nginx.org/download/$_pkgname-$pkgver.tar.gz
...
+        provides="$(eval "echo \$_${name}_provides")"
```
Comment (jirutka) [inline testing/nginx-naxsi/APKBUILD]: You don’t declare any `_<mod>_provides` variable, so this is redundant.

File: testing/nginx-naxsi/APKBUILD [hunk 8]
```diff
@@ -32,119 +51,154 @@ source="http://nginx.org/download/$_pkgname-$pkgver.tar.gz
...
+        depends=
+
+        mkdir -p "$subpkgdir"/usr/share/vim
+        cp -r "$builddir"/contrib/vim "$subpkgdir"/usr/share/vim/vimfiles
+}
+
+_module() {
+        local name="${subpkgname#$pkgname-mod-}"
+        name="${name//-/_}"
+        local soname="$(eval "echo \$_${name}_so")";
```
Comment (jirutka) [inline testing/nginx-naxsi/APKBUILD]: You don’t declare any `_<mod>_so` variable, so this is redundant.

File: testing/nginx-naxsi/APKBUILD (hunk 1 above)
Comment (itoffshore) [inline testing/nginx-naxsi/APKBUILD]: changed

File: testing/nginx-naxsi/APKBUILD (hunk 2 above)
Comment (itoffshore) [inline testing/nginx-naxsi/APKBUILD]: removed

File: testing/nginx-naxsi/APKBUILD (hunk 3 above)
Comment (itoffshore) [inline testing/nginx-naxsi/APKBUILD]: changed

File: testing/nginx-naxsi/APKBUILD (hunk 4 above)
Comment (itoffshore) [inline testing/nginx-naxsi/APKBUILD]: removed

File: testing/nginx-naxsi/APKBUILD (hunk 8 above)
Comment (itoffshore) [inline testing/nginx-naxsi/APKBUILD]: removed

File: testing/nginx-naxsi/APKBUILD (hunk 6 above)
Comment (itoffshore) [inline testing/nginx-naxsi/APKBUILD]: removed

Comment (itoffshore) [inline testing/nginx-naxsi/APKBUILD]: removed

File: testing/nginx-naxsi/APKBUILD (hunk 7 above)
Comment (itoffshore) [inline testing/nginx-naxsi/APKBUILD]: removed

File: testing/nginx-naxsi/nginx.initd (hunk 5 above)
Comment (itoffshore) [inline testing/nginx-naxsi/nginx.initd]: if you set the ip address in `container/config` it breaks `nginx` starting

File: testing/nginx-naxsi/APKBUILD (hunk 3 above)
Comment (itoffshore) [inline testing/nginx-naxsi/APKBUILD]: removed

File: testing/nginx-naxsi/nginx.initd (hunk 5 above)
Comment (itoffshore) [inline testing/nginx-naxsi/nginx.initd]: `need net` left alone - it does however stop `nginx` starting in `lxc` containers with ip addresses set in `container/config`

File: testing/nginx-naxsi/APKBUILD (hunk 3 above)
Comment (itoffshore) [inline testing/nginx-naxsi/APKBUILD]: `main/nginx` sets the temp directories under `/var/lib/nginx/tmp` - I have used the same directory structure

File: testing/nginx-naxsi/APKBUILD (hunk 6 above)
Comment (itoffshore) [inline testing/nginx-naxsi/APKBUILD]: set `depends="!nginx-mod-$name"`

**Comment by itoffshore** [comment]
rebased with changes

File: testing/nginx-naxsi/APKBUILD (hunk 3 above)
Comment (jirutka) [inline testing/nginx-naxsi/APKBUILD]: Hm, I see, and there’s even more weird stuff in that package. I’ll fix it later.

File: testing/nginx-naxsi/nginx.initd (hunk 5 above)
Comment (jirutka) [inline testing/nginx-naxsi/nginx.initd]: Aha, I don’t use statically assigned address, but DHCP. However, there are dozens of runscripts with `need net`, so if this is an issue on LXC, then it must be solved in OpenRC, not just in single runscript. Please open an issue on http://bugs.alpinelinux.org/.

File: testing/nginx-naxsi/APKBUILD [hunk 9]
```diff
@@ -32,119 +51,153 @@ source="http://nginx.org/download/$_pkgname-$pkgver.tar.gz
...
+                --with-http_perl_module=dynamic \
+                --with-mail=dynamic \
+                --with-mail_ssl_module \
+                --with-stream=dynamic \
+                --with-stream_ssl_module \
+                \
+                --add-dynamic-module="$_ngx_naxsi_dir" \
+                --add-dynamic-module="$_ngx_cache_purge_dir" \
+                --add-dynamic-module="$_ngx_upstream_fair_dir" \
+                --add-dynamic-module="$_ngx_http_sysguard_dir" \
```
Comment (jirutka) [inline testing/nginx-naxsi/APKBUILD]: Please use only tabs for indentation, I really don’t enjoy fixing it all the time. ;)

**Comment by itoffshore** [comment]
Sorry thought I fixed them all (I use tabs normally) - these spaces are
from copy & pasting.

Next time just let me know - it's no problem to update a PR

**Pull request closed without merging**